lib/common/entropy_common.c

```diff
@@ -38,7 +38,11 @@ const char* HUF_getErrorName(size_t code) { return ERR_getErrorName(code); }
 /*-**************************************************************
 *  FSE NCount encoding-decoding
 ****************************************************************/
+#if defined(__IAR_SYSTEMS_ICC__)
+MEM_STATIC
```

Comment:
Is there a problem with `FORCE_INLINE_TEMPLATE` ?
Why is it a problem here specifically ?

Reply:
> Is there a problem with `FORCE_INLINE_TEMPLATE` ? Why is it a problem here specifically ?

The IAR compiler complains that the `FSE_readNCount_body` function is too complex to be inlined.

Reply:
The function is indeed pretty big.

It seems the point of `FORCE_INLINE_TEMPLATE` is to pass down `BMI2_TARGET_ATTRIBUTE` compilation directive to an instance of the function, which is invoked when the appropriate target (`AVX2` + `BMI2` architecture) is detected at runtime.

Of course, this capability is irrelevant for `ARM` architectures.

So yes, technically this modification works. It's just a pity that it translates into one exception which will have to be maintained for one compiler.

Reply:
Would it be better to use `HINT_INLINE` instead of `FORCE_INLINE_TEMPLATE`?